src/CoreBundle/Service/PortalDomainService.php

```diff
@@ -121,18 +117,7 @@ private function initialize()
      */
     public function getDefaultPortal()
     {
-        if (null !== $this->defaultPortal) {
-            return $this->defaultPortal;
-        }
-
-        $portalList = TdbCmsPortalList::GetList();
-        $portalList->GoToStart();
-        $this->defaultPortal = $portalList->Current();
-        if (false === $this->defaultPortal) {
-            return null;
-        }
-
-        return $this->defaultPortal;
+        return \TdbCmsConfig::GetInstance()->GetPrimaryPortal();
```

Comment:
please add an update that checks if this will result in a chance for the customer (that is, if the default portal is now different from the first portal).

Reply:
Added an appropriate db update with warning message.